torch/_dynamo/variables/constant.py

```diff
@@ -98,6 +98,10 @@ def call_method(
             if name == "__add__":
                 assert len(args) == 1
                 return args[0].call_method(tx, name, [self], {})
+            # For non-commutative op sub, swap args and invoke rsub.
+            if name == "__sub__":
+                assert len(args) == 1
+                return args[0].call_method(tx, "__rsub__", [self], {})
```

Comment:
fwiw, we should use radd for add as well, because that more closely preserves original semantics

Reply:
Also, there is a clear opportunity to correctly implement this for *all* reversible magic methods, not just sub.